searchlib/src/vespa/searchlib/tensor/tensor_buffer_operations.h

```diff
@@ -0,0 +1,81 @@
+// Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#pragma once
+
+#include <vespa/eval/eval/cell_type.h>
+#include <vespa/vespalib/util/string_id.h>
+#include <cstddef>
+#include <memory>
+
+namespace vespalib {
+template <typename T> class ArrayRef;
+template <typename T> class ConstArrayRef;
+class nbostream;
+}
+
+namespace vespalib::eval {
+struct Value;
+class ValueType;
+}
+
+namespace search::tensor {
+
+/*
+ * Class used to store a tensor in a buffer and make tensor views based on
+ * buffer content.
+ */
```

Comment:
Would be nice with a short explanation of the memory format here as well.

Reply:
Short description added.